applications/GeoMechanicsApplication/python_scripts/geomechanics_analysis.py

```diff
@@ -5,7 +5,9 @@
 sys.path.append(os.path.join('..','..','..'))
 
 import KratosMultiphysics as Kratos
+# Linear solver is not working as expected
 import KratosMultiphysics.LinearSolversApplication
```

Comment:
```suggestion
```

this should not be imported here, but only if used (handled by the `python_linear_solver_factory`)

Reply:
By removing this, I get an error from python_linear_solver_factory.py saying:

RuntimeError: Error: Trying to construct a Linear solver with solver_type:
"sparse_lu" which does not exist.
The list of available options (for currently loaded applications) is:
Kratos components
    amgcl
    amgcl_ns
    bicgstab
    cg
    deflated_cg
    gmres
    monotonicity_preserving
    scaling
    skyline_lu_factorization
    super_lu
    super_lu_iterative
    tfqmr

Reply:
if you want to use solvers from an app you should specify this in the name.
In your case you should put `LinearSolversApplication.sparse_lu`

=> this way it is extensible in case in the future there will be other applications
Example: GPUSolversApplication

If you hardcode the import of those apps in the stage, then this can be quite an inconvenience for the users without benefit
e.g. now the users have to compile the ExternalSolversApp even if they don't use it (and compiling it can be quite a pain, esp in Win)

see also https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/python_scripts/python_linear_solver_factory.py#L16-L24

Reply:
Thanks! Resolved.